app/main.py

```diff
@@ -1,3 +1,16 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Any, Callable
+
+
+def cache(func: Callable) -> Callable:
+    runs_data = {}
+
+    def saver(*args) -> Any:
+        if args in runs_data:
+            print("Getting from cache")
+            return runs_data[args]
+        else:
+            print("Calculating new result")
+            runs_data[args] = func(*args)
+            return runs_data[args]
```

Comment:
with those `return`s you have code duplication. In general, it is not a very good practice. Try to fix this issue

Reply:
Oh, I see. Thank you!